cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -74,22 +73,19 @@ runTxSign (TxBodyFile infile) skfiles  network (TxFile outfile) = do
       . writeTxSigned outfile
       $ signTransaction txu network sks
 
-runTxSubmit :: FilePath -> ConfigYamlFilePath -> SocketPath -> ExceptT CliError IO ()
-runTxSubmit txFp configFp sktFp =
+runTxSubmit :: FilePath -> Network -> SocketPath -> ExceptT CliError IO ()
+runTxSubmit txFp network sktFp =
   withIOManagerE $ \iocp -> do
-    nc <- liftIO $ parseNodeConfigurationFP configFp
-    SomeConsensusProtocol p <- firstExceptT ProtocolError $ mkConsensusProtocol nc Nothing
     signedTx <- firstExceptT CardanoApiError . newExceptT $ readTxSigned txFp
-    case p of
-      Consensus.ProtocolRealTPraos{} -> do
-        let config = pInfoConfig $ Consensus.protocolInfo p
-        liftIO $ submitTx
-                   nullTracer -- tracer needed
-                   iocp
-                   config
-                   sktFp
-                   (prepareTxShelley signedTx)
-      _ -> left $ IncorrectProtocolSpecifiedError (ncProtocol nc)
+
+    let config = Consensus.protocolClientInfo mkNodeClientProtocolTPraos
```

Comment:
This stuff should be pushed back into the `submitTransaction` function in `cardano-api`

This function should only read the file and call `submitTransaction`.